sdk/core/allocator/revoker.h

```diff
@@ -87,52 +120,69 @@ namespace Revocation
 		 */
 		void shadow_paint_range(ptraddr_t base, ptraddr_t top, bool fill)
 		{
-			constexpr size_t ShadowWordAddrMask =
-			  (1U << (ShadowWordShift + MallocAlignShift)) - 1;
-			ptraddr_t baseUp =
-			  (base + ShadowWordAddrMask) & ~ShadowWordAddrMask;
-			ptraddr_t topDown = top & ~ShadowWordAddrMask;
-
-			// There isn't a single aligned shadow word for this range, so paint
-			// one bit at a time.
-			if (baseUp >= topDown)
+			size_t baseCapOffset = shadow_offset_bits(base);
```

Comment:
A very small increase and a small perf gain in the test suite; we just don't free that many objects.  See https://github.com/microsoft/cheriot-rtos/pull/52/commits/f622b87d717f957a562a7ca4dfd16d7a3b14f8df

Reply:
I wonder if just doing `__always_inline` without the `template` would lead to the same results.

Reply:
Oddly, no, in that `__always_inline`-ing by itself is somehow both a little faster
```
Test runner: Allocator finished in 6239046 cycles
Test runner: All tests finished in 6638386 cycles
```
and a little larger
```
   text    data     bss     dec     hex filename
  92718    2940    6255  101913   18e19 tests/build.sail/cheriot/cheriot/release/test-suite
```
than the other options.  I would have thought that templating and constant propagation would have been equivalent here, but I guess we're tickling slightly different pattern matches during optimization or something like that.

Since the instruction gains are so small and space is at a premium, I'm tempted to drop this for the moment, unless there's clear reason to go with another option... @davidchisnall?

Reply:
Smaller is probably better than slightly faster.

Reply:
OK; dropped that commit.  Still happy with the rest?